src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -435,9 +439,114 @@ public static boolean isGenomicsDBPath(final String path) {
             exportConfigurationBuilder.setGenerateArrayNameFromPartitionBounds(true);
         }
 
+        //Sample code snippet to show how combine operations for INFO fields can be specified using the Protobuf
+        //API
+        //
+        //References
+        //GenomicsDB Protobuf structs: https://github.com/Intel-HLS/GenomicsDB/blob/master/src/resources/genomicsdb_vid_mapping.proto
+        //Protobuf generated Java code guide:
+        //https://developers.google.com/protocol-buffers/docs/javatutorial#the-protocol-buffer-api
+        //https://developers.google.com/protocol-buffers/docs/reference/java-generated
+
+        //Parse the vid json and create an in-memory Protobuf structure representing the
+        //information in the JSON file
+        GenomicsDBVidMapProto.VidMappingPB vidMapPB;
+        try {
+            vidMapPB = getProtobufVidMappingFromJsonFile(vidmapJson);
+        } catch (final IOException e) {
+            throw new UserException("Could not open vid json file " + vidmapJson, e);
+        }
+
+        //In vidMapPB, fields is a list of GenomicsDBVidMapProto.GenomicsDBFieldInfo objects
+        //Each GenomicsDBFieldInfo object contains information about a specific field in the TileDB/GenomicsDB store
+        //We iterate over the list and create a field name to list index map
+        final HashMap<String, Integer> fieldNameToIndexInVidFieldsList =
+                getFieldNameToListIndexInProtobufVidMappingObject(vidMapPB);
+
+        //Example: set MQ combine operation to median (default is also median, but this is just an example)
+        vidMapPB = updateINFOFieldCombineOperation(vidMapPB, fieldNameToIndexInVidFieldsList,
+                "MQ", "median");
+        if (vidMapPB != null) {
+            //Use rebuilt vidMap in exportConfiguration
+            //NOTE: this does NOT update the JSON file, the vidMapPB is a temporary structure that's passed to
+            //C++ modules of GenomicsDB for this specific query. Other queries will continue to use the information
+            //in the JSON file
+            exportConfigurationBuilder.setVidMapping(vidMapPB);
+        }
+
         return exportConfigurationBuilder.build();
     }
 
+    /**
+     * Parse the vid json and create an in-memory Protobuf structure representing the
+     * information in the JSON file
+     *
+     * @param vidmapJson vid JSON file
+     * @return Protobuf object
+     */
+    public static GenomicsDBVidMapProto.VidMappingPB getProtobufVidMappingFromJsonFile(final File vidmapJson)
```

Comment:
@ldgauthier Can you pull out these GenomicsDB-specific utility methods (`createExportConfiguration()`, `getProtobufVidMappingFromJsonFile()`, etc. into a separate `GenomicsDBUtils` class, instead of cluttering `FeatureDataSource` with them.

Could you also update Karthik's comments to no longer say things like "Sample code snippet", and remove commented-out code?